scripts/thumbnail-generator/generate_thumbnail.py

```diff
@@ -0,0 +1,302 @@
+    translateOp = None
+    # Go through each operation in the xformable schema
+    for op in xformRoot.GetOrderedXformOps():
+        # If the operation is a translate operation, we've found our operation
+        if op.GetOpType() == UsdGeom.XformOp.TypeTranslate:
+            translateOp = op
+            break
+
+    # If no translate operation exists, create one
+    if translateOp is None:
+        translateOp = xformRoot.AddTranslateOp(UsdGeom.XformOp.PrecisionDouble)
+
+    translateOp.Set(camera_translation)
+    camera_stage.Save()
+
+def sublayer_subject(camera_stage, input_file):
+    camera_stage.GetRootLayer().subLayerPaths = [input_file]
+    camera_stage.GetRootLayer().Save()
+
+def take_snapshot(image_name):
+    renderer = get_renderer()
+    cmd = ['usdrecord', '--camera', 'MainCamera', '--imageWidth', str(args.width), '--renderer', renderer, 'camera.usda', image_name]
+    run_os_specific_usdrecord(cmd)
+    os.remove("camera.usda")
+    if os.path.isfile("y_up.usda"):
+        os.remove("y_up.usda")
+    return image_name
+
+def get_renderer():
+    if os.name == 'nt':
+        print("windows default renderer GL being used...")
+        return "GL"
+    else:
+        if sys.platform == 'darwin':
+            print("macOS default renderer Metal being used...")
+            return 'Metal'
+        else:
+            print("linux default renderer GL being used...")
+            return 'GL'
+
+def run_os_specific_usdrecord(cmd):
+    if os.name == 'nt':
+        subprocess.run(cmd, check=True, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+    else:
+        if sys.platform == 'darwin':
+            subprocess.run(cmd, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+        else:
+            subprocess.run(cmd, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+def create_image_filename(input_path, extension):
+    return Path(input_path).with_suffix("." + extension)
+
+def link_image_to_subject(subject_stage, image_name):
+    subject_root_prim = subject_stage.GetDefaultPrim()
+    mediaAPI = UsdMedia.AssetPreviewsAPI.Apply(subject_root_prim)
+    thumbnails = UsdMedia.AssetPreviewsAPI.Thumbnails(defaultImage = Sdf.AssetPath(image_name))
+    mediaAPI.SetDefaultThumbnails(thumbnails)
+    subject_stage.GetRootLayer().Save()
+    
+def create_usdz_wrapper_stage(usdz_file):
+    file_name = usdz_file.split('.')[0]
+    existing_stage = Usd.Stage.Open(usd_file)
+    new_stage = Usd.Stage.CreateNew(file_name + '_Thumbnail.usda')
+    
+    UsdUtils.CopyLayerMetadata(existing_stage.GetRootLayer(), new_stage.GetRootLayer())
+
+    new_stage.GetRootLayer().subLayerPaths = [usdz_file]
+    new_stage.GetRootLayer().Save()
+    return new_stage
+
+def zip_results(usd_file, image_name, is_usdz):
+    file_list = [usd_file, image_name]
+    usdPath = Path(usd_file)
+
+    if is_usdz:
+        file_list.append(usdPath.with_suffix('_Thumbnail.usda'))
```

Comment:
Repeated use of hard-coded string value (_Thumbnail.usda) should be replaced by a constant variable (`THUMBNAIL_LAYER_SUFFIX = "_Thumbnail.usda"`).